src/aktualizr_repo/image_repo.cc

```diff
@@ -73,6 +73,43 @@ void ImageRepo::addDelegation(const Uptane::Role &name, const std::string &path,
   updateRepo();
 }
 
+void ImageRepo::revokeDelegation(const Uptane::Role &name) {
+  if (keys_.count(name) == 0) {
+    throw std::runtime_error("Delegation does not exist.");
+  }
+  if (Uptane::Role::IsReserved(name.ToString())) {
+    throw std::runtime_error("Delegation name " + name.ToString() + " is reserved.");
+  }
+
+  boost::filesystem::path keys_dir = path_ / ("keys/image/" + name.ToString());
+  boost::filesystem::remove_all(keys_dir);
+
+  boost::filesystem::path repo_dir(path_ / "repo/image");
+  boost::filesystem::remove(repo_dir / (name.ToString() + ".json"));
```

Comment:
Should revoke delete the delegated repository? In the classic scenario, revoked repos should be ignored even if still reachable via network, by signing new delegating metadata without it.

Reply:
Yeah, but why it's needed? We couldn't restore it anymore, and it will be overwritten if new delegation with the same name is created.

Reply:
I believe tests would be more robust if we actually check that delegations are ignored even if still reachable. But that's fine, we can still change that later if needed.